DXRModules/DeusEx/Classes/DXRAugmentations.uc

```diff
@@ -524,7 +569,82 @@ static simulated function string DescriptionLevelExtended(Actor act, int i, out
     }
 }
 
-simulated function RemoveRandomAug(#var(PlayerPawn) p)
+simulated function CleanUpAugCounts(#var(PlayerPawn) p)
+{
+    local int i;
+    local Augmentation a;
+    local AugmentationManager am;
+
+    am = p.AugmentationSystem;
+
+    //Clear out the existing augCounts in each slot, since the slots were randomized...
+    for (i=0;i<ArrayCount(am.AugLocs);i++){
+        am.AugLocs[i].augCount=0;
+    }
+
+    //Recount the official augCount per aug slot
+    for (a=am.FirstAug; a!=None; a = a.next ) {
+        if (!a.bHasIt) continue;
+        am.AugLocs[a.AugmentationLocation].augCount++;
+    }
+}
+
+simulated function int CleanUpAugSlots(#var(PlayerPawn) p)
+{
+    local int numAugs[7]; //Count of the number of augs in each slot
+    local int augsRemoved, i, j;
+    local Augmentation a;
+    local AugmentationManager am;
+
+    am = p.AugmentationSystem;
+
+    //Get a count of how many augs you have in each slot
+    for (a=am.FirstAug; a!=None; a = a.next ) {
+        if (!a.bHasIt) continue;
+        if (a.AugmentationLocation == LOC_Default) continue;
+        numAugs[a.AugmentationLocation]++;
```

Comment:
do we really need this when CleanUpAugCounts fixes am.AugLocs[a.AugmentationLocation].augCount